tensorflow/contrib/seq2seq/python/ops/decoder_fn.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2016 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Seq2seq loss operations for use in neural networks.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from tensorflow.python.ops import array_ops
+
+
+__all__ = ["decoder_fn_train",
+           "decoder_fn_eval_wrapper"]
+
+def decoder_fn_train(encoder_state):
+
+  def decoder_fn(time, cell_state, cell_output, context_state):
+    if cell_state == None: # first call, return encoder_state
+      return (None, encoder_state, None, cell_output, context_state)
+    else:
+      return (None, cell_state, None, cell_output, context_state)
+  return decoder_fn
+
+
+def decoder_fn_eval(encoder_state, embeddings, sos_id, eos_id, max_len, dtype=tf.int32):
```

Comment:
use more explicit naming here, `sos_id` doesn't mean much for the general practitioner.  always use full names, e.g. for sos_id/ eos_id, and e.g. max_len should be `maximum_length`.

also provide a full docstring.